test/src/main/java/org/apache/accumulo/test/functional/CloneTestIT.java

```diff
@@ -350,4 +353,64 @@ public void testCloneMetadataTable() {
           "mc1", CloneConfiguration.empty()));
     }
   }
+
+  @Test
+  public void testCloneIntoDiffNamespace() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+      client.tableOperations().create(tableName);
+      client.namespaceOperations().create("new");
+      client.tableOperations().clone(tableName, "new." + tableName, CloneConfiguration.empty());
+    }
+  }
+
+  @Test
+  public void testCloneIntoDiffNamespaceDoesntExist() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+      client.tableOperations().create(tableName);
+      assertThrows(AccumuloException.class, () -> client.tableOperations().clone(tableName,
+          "missing." + tableName, CloneConfiguration.empty()));
+    }
+  }
+
+  @Test
+  public void testCloneIntoAccumuloNamespace() throws Exception {
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+      client.tableOperations().create(tableName);
+      assertThrows(AccumuloException.class, () -> client.tableOperations().clone(tableName,
+          "accumulo." + tableName, CloneConfiguration.empty()));
+    }
+  }
+
+  @Test
+  public void testCloneNamespacePermissions() throws Exception {
+    final String tableName = getUniqueNames(1)[0];
+    final String newUserName = "NEW_USER";
+    final String srcNs = "src";
+    final String srcTableName = srcNs + "." + tableName;
+    final String destNs = "dst";
+    final String dstTableName = destNs + "." + tableName;
+
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      client.namespaceOperations().create(srcNs);
+      client.tableOperations().create(srcTableName);
+      client.namespaceOperations().create(destNs);
+      client.securityOperations().createLocalUser(newUserName, new PasswordToken(newUserName));
+      // User needs WRITE or ALTER_TABLE on the src table to flush it as part of the clone operation
+      client.securityOperations().grantTablePermission(newUserName, srcTableName,
+          TablePermission.ALTER_TABLE);
+      client.securityOperations().grantTablePermission(newUserName, srcTableName,
+          TablePermission.READ);
```

Comment:
This is the expected success case. However, prior to this fix, I believe if this was set on the destination namespace, but *NOT* on the source table, then the operation would still succeed. The test should check that case, to make sure that it succeeds before this fix (to verify the bug), and fails after this fix.

Reply:
Added suggested test in 3cfb06c. I will see if it succeeds in the 2.1 branch.

Reply:
I copied [this](https://github.com/apache/accumulo/pull/5334/files#diff-89ff01550cac0c44b957f8c13e910aed15aee2f419bcdcb825517ebefdf22a94R425) test to the 2.1 branch and is succeeds as is with no modifications. Based on what you are saying above, this test should have failed because the clone operation should have succeeded instead of throwing an exception.

Reply:
I copied the test and it failed on the assertThrows... (the clone succeeded with the wrong permissions).

Reply:
I also confirmed it passes in your branch after the fixes.

Reply:
> I copied [this](https://github.com/apache/accumulo/pull/5334/files#diff-89ff01550cac0c44b957f8c13e910aed15aee2f419bcdcb825517ebefdf22a94R425) test to the 2.1 branch and is succeeds as is with no modifications. Based on what you are saying above, this test should have failed because the clone operation should have succeeded instead of throwing an exception.

@dlmarion what's the status of this PR? Based on this comment, it seems like you were saying the problem isn't reproducible in 2.1. However, your test reproduces it for me in 2.1, and this PR seems to fix it. So... I think this PR is ready.

Reply:
IIRC, I was waiting on someone to tell me what, if any, changes needed to occur to the table / namespace locking based on the conversation at https://github.com/apache/accumulo/pull/5334#discussion_r1956430797. It wasn't clear to me based on the comments what changes I needed to make.

Reply:
Okay, I would also feel more comfortable having @keith-turner take a look at the fate locks to make sure they make sense. Otherwise, I think this can be merged.

I was also concerned that you were not able to reproduce the problem in 2.1, because I was able to reproduce it using your test.

Reply:
I backported [this](https://github.com/apache/accumulo/pull/5334/files#diff-89ff01550cac0c44b957f8c13e910aed15aee2f419bcdcb825517ebefdf22a94R425) test to 2.1 and it fails now, so it looks like I can reproduce the issue. Odd that it didn't fail for me back on 14 Feb, maybe I did something wrong.